install-jellyfin.ps1

```diff
@@ -0,0 +1,385 @@
+        mkdir $Script:DefaultJellyFinInstallDirectory
+    }
+    Copy-Item -Path $PSScriptRoot/* -DestinationPath "$Script:DefaultJellyFinInstallDirectory/" -Force -Recurse
+    if($Script:InstallAsService){
+        if($Script:InstallServiceAsUser){
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" install Jellyfin `"$Script:DefaultJellyFinInstallDirectory\jellyfin.exe`"
+            Start-Sleep -Milliseconds 500
+            &sc.exe config Jellyfin obj=".\$($Script:UserCredentials.UserName)" password="$($Script:UserCredentials.GetNetworkCredential().Password)"
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" set Jellyfin Start SERVICE_DELAYED_AUTO_START 
+        }else{
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" install Jellyfin `"$Script:DefaultJellyFinInstallDirectory\jellyfin.exe`"
+            Start-Sleep -Milliseconds 500
+            #&"$Script:DefaultJellyFinInstallDirectory\nssm.exe" set Jellyfin ObjectName $Script:UserCredentials.UserName $Script:UserCredentials.GetNetworkCredential().Password
+            #Set-Service -Name Jellyfin -Credential $Script:UserCredentials
+            Start-Sleep -Milliseconds 500
+            &"$Script:DefaultJellyFinInstallDirectory\nssm.exe" set Jellyfin Start SERVICE_DELAYED_AUTO_START 
+        }
+    }
+    if($Script:MigrateLibrary){
+        Copy-Item -Path $Script:defaultEmbyDataDir/config -Destination $Script:JellyfinDataDir -force -Recurse
+        Copy-Item -Path $Script:defaultEmbyDataDir/cache -Destination $Script:JellyfinDataDir -force -Recurse
+        Copy-Item -Path $Script:defaultEmbyDataDir/data -Destination $Script:JellyfinDataDir -force -Recurse
+        Copy-Item -Path $Script:defaultEmbyDataDir/metadata -Destination $Script:JellyfinDataDir -force -Recurse
+        Copy-Item -Path $Script:defaultEmbyDataDir/root -Destination $Script:JellyfinDataDir -force -Recurse
+    }
+    if($Script:CreateShortcut){
+        $WshShell = New-Object -comObject WScript.Shell
+        $Shortcut = $WshShell.CreateShortcut("$Home\Desktop\Jellyfin.lnk")
+        $Shortcut.TargetPath = "$Script:DefaultJellyFinInstallDirectory\jellyfin.exe"
+        $Shortcut.Save()
+    }
+    if($Script:StartJellyfin){
+        if($Script:InstallAsService){
+            Get-Service Jellyfin | Start-Service
+        }else{
+            Start-Process -FilePath $Script:DefaultJellyFinInstallDirectory\jellyfin.exe -PassThru
+        }
+    }
+}else{
+
+}
+Add-Type -AssemblyName System.Windows.Forms
+[System.Windows.Forms.Application]::EnableVisualStyles()
+
+$Script:JellyFinDataDir = "$env:AppData\jellyfin\"
+$Script:DefaultJellyFinInstallDirectory = "$env:Appdata\jellyfin\"
+$Script:defaultEmbyDataDir = "$env:Appdata\Emby-Server\"
+$Script:InstallAsService = $False
+$Script:InstallServiceAsUser = $false
+$Script:CreateShortcut = $false
+$Script:MigrateLibrary = $false
+$Script:StartJellyfin = $false
+
+function InstallJellyfin {
```

Comment:
This seems to duplicate code above, please explain

Reply:
The section above will only run if you pass in the -quiet parameter, this section is what runs if you launch it for a GUI. Ideally they'd be merged into a single function with parametersets and a switch statement for them, but they aren't because I was trying to get it finished.

Reply:
A "fix me" comment would be f8ne with me for now. First we get it working, then we'll make it pretty :)